pennylane/ops/arithmetic/sum.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Sum class which contains logic for
+computing the sum of operations.
+"""
+import warnings
+import numpy as np
+import pennylane as qml
+from pennylane import numpy as qnp
+
+
+def sum(*summands):
+    """Top level function computing the sum of the
+    provided summands"""
+    return Sum(*summands)
+
+
+class Sum(qml.operation.Operator):
+    """Arithmetic operator subclass representing the sum of operators"""
+
+    def __init__(self, *summands, do_queue=True, id=None):
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
+
+        self.summands = summands
+
+        combined_wires = qml.wires.Wires.all_wires([s.wires for s in summands])
+        combined_params = [s.parameters for s in summands]
+        super().__init__(
+            *combined_params, wires=combined_wires, do_queue=do_queue, id=id
+        )
+        self._name = "Sum(" + ", ".join([f"{f}" for f in summands]) + ")"
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " + ".join([f"{f}" for f in self.summands])
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    def terms(self):
+        return [1.0]*len(self.summands), self.summands
+
+    def get_matrix(self):
```

Comment:
Remember that `get_matrix` takes a `wire_order`.

We could think about overriding `compute_matrix` instead, but I think we may need to override `get_matrix` instead for Sum, since we need the wire order.

Reply:
Yes, that makes things a lot cleaner! I tried to overwrite the compute methods before, which led to very ugly constructions. And this was kind of envisioned, that `compute_matrix` is more a lower-level function to define a numerical representation, while here we want to just define an operation on numerical representations.

So it should be 

```def get_matrix(wire_order=None):

         if wire_order is None:
                  wire_order = self.wires
         .... m.get_matrix(wire_order=wire_order)

```

Sorry, thought that was what I did in the prototype!

Reply:
Addressed this 👍🏼